types/exorcist/index.d.ts

```diff
@@ -1,21 +1,88 @@
 
-    /** Externalizes the source map found inside a stream to an external .map file.
-     * Works with both JavaScript and CSS input streams
-     * @param file full path to the map file to which to write the extracted source map
-     * @param [url] full URL to the map file, set as sourceMappingURL in the streaming output (default: file)
-     * @param [root] root URL for loading relative source paths, set as sourceRoot in the source map (default: "")
-     * @param [base] base path for calculating relative source paths (default: use absolute paths)
-     * @param [errorOnMissing] when truthy, causes 'error' to be emitted instead of 'missing-map' if no map was found in the stream (default: falsey)
-     */
-    function exorcist(file: string, url?: string, root?: string, base?: string, errorOnMissing?: boolean): through.ThroughStream;
-    export = exorcist;
+export = exorcist;
+
+/**
+ * Externalizes the source map of the file streamed in.
+ *
+ * The source map is written as JSON to `input`, and the original file is streamed out with its
+ * `sourceMappingURL` set to the path of `input` (or to the value of `url`).
+ *
+ * @param input file path or writable stream where the source map will be written
+ * @param url full URL to the map file, set as `sourceMappingURL` in the streaming output (default: `input`)
+ * @param root root URL for loading relative source paths, set as `sourceRoot` in the source map (default: "")
+ * @param base base path for calculating relative source paths (default: use absolute paths)
+ * @param errorOnMissing when truthy, causes `"error"` to be emitted instead of `"missing-map"` if no map was found in the stream (default: falsy)
+ * @returns transform stream into which to pipe the code containing the source map
+ *
+ * @example
+ * import browserify = require("browserify");
+ * import * as path from "path";
+ * import * as fs from "fs";
+ * import exorcist = require("exorcist");
+ *
+ * const mapfile = path.join(__dirname, "bundle.js.map");
+ *
+ * // from a file, to a file, and send source map to its own file
+ * browserify({ debug: true })
+ *     .require(require.resolve("./main"), { entry: true })
+ *     .bundle()
+ *     .pipe(exorcist(mapfile))
+ *     .pipe(fs.createWriteStream(path.join(__dirname, "bundle.js"), "utf8"));
+ *
+ * // from a stream, to a stream, and send source map to a stream
+ * browserify([readableSourceStream], browserifyOptions)
+ *     .bundle()
+ *     .pipe(exorcist(targetSourceMapStream, "/url/path/to/replace/source/comment/with/bundle.js"))
+ *     .pipe(writableTargetStream);
+ */
+declare function exorcist(
+    input: string,
+    url?: string,
+    root?: string,
+    base?: string,
+    errorOnMissing?: boolean,
+): exorcist.Stream;
+declare function exorcist(
+    input: Writable,
+    url: string,
+    root?: string,
+    base?: string,
+    errorOnMissing?: boolean,
+): exorcist.Stream;
+
+declare namespace exorcist {
+    type Stream = ThroughStream & {
```

Comment:
Interesting, I guess these custom 'event' "missing-map" types are for newer version of TypeScript so users get 'event' auto-completion in their preferred code editor, and for the custom documentation comments?

Reply:
exactly